p126.java

```diff
@@ -0,0 +1,90 @@
+/*
+126. Word Ladder II
+
+Given two words (beginWord and endWord), and a dictionary's word list, find all shortest transformation sequence(s) from beginWord to endWord, such that:
+
+Only one letter can be changed at a time
+Each transformed word must exist in the word list. Note that beginWord is not a transformed word.
+For example,
+
+Given:
+beginWord = "hit"
+endWord = "cog"
+wordList = ["hot","dot","dog","lot","log","cog"]
+
+Return
+  [
+    ["hit","hot","dot","dog","cog"],
+    ["hit","hot","lot","log","cog"]
+  ]
+*/
+
+public class Solution {
+
+    public List<List<String>> findLadders(String beginWord, String endWord, List<String> wordList) {
+        Map<String, Integer> map = new HashMap<>();
+        Map<String, Set<String>> neighbors = new HashMap<>();
+        neighbors.put(beginWord, new HashSet<>());
+        map.put(beginWord, -1);
+        Set<String> wordSet = new HashSet<>();
+        for(String word : wordList){
+            wordSet.add(word);
+            map.put(word, -1);
+            neighbors.put(word, new HashSet<>());
+        }
+        int distance = 0;
+        boolean found = false;
+        Queue<String> queue = new LinkedList<>();
+        queue.offer(beginWord);
+        while(!queue.isEmpty()){
+            int len = queue.size();
+            for(int i = 0; i < len; i++){
+                String curWord = queue.poll();
+                if(map.get(curWord) == -1 || map.get(curWord) > distance){
```

Comment:
This condition can be simplified. Think about if such scenario exists: you visit a node, and the node has been visited and it has a larger distance than current distance. Is that possible?